src/form-fields/DatePicker.component.js

```diff
@@ -6,14 +6,20 @@ import { DatePicker as MuiDatePicker } from 'material-ui';
 class DatePicker extends React.Component {
     constructor(props) {
         super(props);
-        this.maxDate = props.allowFuture ? undefined : new Date();
-        this.props = props;
-        this._onDateSelect = this._onDateSelect.bind(this);
-        this._formatDate = this._formatDate.bind(this);
+        const {
+            allowFuture,
+            dateFormat,
+            ...other
+        } = props;
+        this.other = other;
```

Comment:
This props magic should probably go in `render()` instead of in the constructor :)